app/frontend/utils/formio-helpers.ts

```diff
@@ -11,6 +11,39 @@ export const singleRequirementFormJson = (requirementJson: IFormIORequirement):
   }
 }
 
+export const singleRequirementSubmissionData = (submissionData: any, requirementKey: string) => {
+  const sectionKey = Object.keys(submissionData.data).find((sk) => submissionData.data[sk][requirementKey])
+  if (!sectionKey) return { data: { [requirementKey]: null } }
+
+  return { data: { [requirementKey]: submissionData.data[sectionKey][requirementKey] } }
+}
+
 export const isFieldSetKey = (key: string) => {
   return key.split("|").length === 4
 }
+
+export const compareSubmissionData = (beforeSubmissionData: ISubmissionData, afterSubmissionData: ISubmissionData) => {
+  const changedFields: string[] = []
+
+  // Helper function to compare two values
+  const isDifferent = (value1: any, value2: any): boolean => {
+    return JSON.stringify(value1) !== JSON.stringify(value2)
+  }
```

Comment:
maybe use https://ramdajs.com/docs/#equals ? `JSON.stringify` is not reliable for comparison. Unless key ordering also matters